Framework/DataHandling/src/MaskDetectors.cpp

```diff
@@ -502,11 +502,11 @@ void MaskDetectors::appendToDetectorListFromWS(std::vector<detid_t> &detectorLis
 
   for (size_t i = 0; i < maskWs->getNumberHistograms(); ++i) {
     if (maskWs->y(i)[0] == 0) {
-      const auto &spec = maskWs->getSpectrum(i);
-      for (const auto &id : spec.getDetectorIDs()) {
-        if (detMap.at(id) >= startIndex && detMap.at(id) <= endIndex)
-          detectorList.emplace_back(id);
-      }
+      const auto &detIds = maskWs->getSpectrum(i).getDetectorIDs();
+      auto condition = [detMap, startIndex, endIndex](const auto id) {
```

Comment:
```suggestion
      auto condition = [detMap, startIndex, endIndex](const auto &id) {
```
You could pass here `id` by reference.

Reply:
Yeah, it is possible but generally passing small types by value like `int` is ok because the reference variable size is similar to `int` size in the memory representation so it will not make a big performance improvement